### ISSUE TITLE
Test skipped if "describe" or "it" is not a string

### ISSUE THREAD

**Issue by andreviallon** [firsttris/vscode-jest-runner#241]
Hello,

I have an issue when running tests in a file where either the name of a "describe" or a "it" is not a string but instead a function.name. In the example bellow I use sum.name as the name for the describe (line 3 in sum.spec.ts) and all tests will be skipped when running tests through the extension.

Does not work:
<img width="1149" alt="Screenshot 2022-03-04 at 13 50 43" src="https://user-images.githubusercontent.com/17278765/156767039-5bd391bc-44df-49c0-9e04-6fb12c3dc9b3.png">
 
Works:
<img width="1155" alt="Screenshot 2022-03-04 at 13 51 15" src="https://user-images.githubusercontent.com/17278765/156767062-274b4173-6789-4837-b15c-2b3a4fcb527b.png">

**Comment by mhombach** [comment]
I can confirm, for us the test is skipped if the test-title is contains something like this with those special quotes inside
```js
it('should trigger `onChange` event when value changes', fakeAsync(() => {...
```
The `onChange` in this case is considered as a bash-command (which doesn't exist) because it's used directly in the commandline.
I think this should be listed in this already existing issue, but if you think it should be a separate one, then I can move it to a new one.

**Comment by AbrahamLopez10** [comment]
We're having the same issue, if this could be fixed it would be awesome, as using the function names provides a much easier and more consistent way of naming test suites for a given method, and makes renaming the test seamless when the method name changes, as it's just using the reference to the method name.

**Comment by mhombach** [comment]
Does the jest-cli support escaping currently? If not, this bug might be needed to be fixed in the jest-cli, not vscode :/

**Comment by andreviallon** [comment]
> Does the jest-cli support escaping currently? If not, this bug might be needed to be fixed in the jest-cli, not vscode :/

Well, the extension works when running tests for the whole file in watch mode, so I guess it's bug

**Comment by mhombach** [comment]
> > Does the jest-cli support escaping currently? If not, this bug might be needed to be fixed in the jest-cli, not vscode :/
> 
> Well, the extension works when running tests for the whole file in watch mode, so I guess it's bug

Testing the full file is not a problem. The problem is, that, when testing specific tests from a test-file, the name of the test-case is passed as parameter to the jest-cli. If this name however contains certain quotation-chars, this is basically ending the parameter too early. I'm not sure if the jest-cli currently supports escaped quotes in its cli-params? I've had this problem with different vscode-extensions, they all had trouble with certain quotes :/

**Comment by domsleee** [comment]
I think `.name` is supported by #301?

The code replaces the `.name` suffix with the empty string here:

https://github.com/firsttris/vscode-jest-runner/blob/b5097b30d55e493203fbc15df37898df48581594/src/util.ts#L101-L102

Please re-open if I'm mistaken 👍

**Comment by nderevj** [comment]
@firsttris 

This issue seems to have resurfaced recently (I'd say within the last month or so).

Tests like the following are skipped when executed through VSCode's inline "run":

```
const sum = require("./sum");

describe(sum.name, () => {
  test("adds 1 + 2 to equal 3", () => {
    expect(sum(1, 2)).toBe(3);
  });
});
```

Extension Version: 0.4.125
Jest Version: 30.2.0

**Comment by firsttris** [comment]
@nderevj 
The issue was caused by the parser not resolving sum.name correctly, as it doesn't execute require or import statements during static analysis. It fell back to using the literal string "sum.name", while Jest registered the test as "sum".

I've added a heuristic to the parser: if it sees Identifier.name on an unresolved identifier, it now assumes the value is the identifier's name itself (e.g., "sum"). This aligns with Function.name behavior and fixes the issue for both describe and it blocks.

Thanks for reporting!
